src/vsr/replica.zig

```diff
@@ -2716,7 +2717,13 @@ pub fn ReplicaType(
             message.header.set_checksum_body(message.body());
             message.header.set_checksum();
 
-            assert(message.header.view <= self.view);
+            // Pings advertise checkpoints, and current checkpoint's view might be greater than
+            // the replica view.
+            if (message.header.view > self.view) {
```

Comment:
Should this be `> self.view_durable()`? We might have just called `transition_to_normal_from_recovering_head_status()` so our `view_durable` is not yet updated, but our `self.view` is updated.

Reply:
I don't think so. I believe that

```
            if (message.header.view > self.view_durable()) {
                assert(self.superblock.working.vsr_state.sync_view >
                    self.superblock.working.vsr_state.view);
            }
```

would hold, but that's an almost tautological assertion. 

In contrast

```
            if (message.header.view > self.view_durable()) {
                assert(self.status == .recovering_head);
            }
```

would _not_ hold -- `transition_to_normal_from_recovering_head_status` changes status, but leaves `self.superblock.working.vsr_state` intact until durable update finishes. 


And

```
            if (message.header.view > self.view) {
                assert(self.status == .recovering_head);

```

Is the interesting case -- it ensures that we can't get out out `.recovering_head` state earlier than `self.superblock.working.vsr_state.view`

Reply:
Ahh I get it; thank you.